### ISSUE TITLE
Bug: Wrong shape of reloaded FRB dataset

### ISSUE THREAD

**Issue by mabruzzo** [yt-project/yt#2858]
### Bug report

**Bug summary**

I think that I stumbled onto a bug when loading a `FixedResolutionBuffer` from a reloadable dataset (although it's possible that I'm doing something dumb). It appears that yt incorrectly permutes the shape of the array data (it's only relevant manifests when the number of elements per axis are unequal). 

Invoking the the `reshape` method seems to resolve the issue. Moreover, inspection of the FRB dataset on the hdf5 file (see the code snippet) reveals that the data is accurately saved to the hdf5 file. Therefore, I suspect that the root issue lies in either the metadata saved by `FixedResolutionBuffer`'s `save_as_dataset` method OR in the yt-data frontend.

**Code for reproduction**

While this code uses Generic Array Data, I first stumbled onto this problem using `enzo_p` data. The following snippet basically validates the description I provided above.

```python
import yt
import numpy as np
import h5py

# construct an arbitrary dataset
arr = np.arange(8.*9.*10.).reshape((8,9,10))
data = dict(density = (arr, "g/cm**3"))
bbox = np.array([[-4,4.],[-4.5,4.5],[-5.,5]])
ds = yt.load_uniform_grid(data, arr.shape, length_unit="Mpc", bbox=bbox,
                          periodicity = (False, False, False))

# make a slice
slc = ds.slice(axis = 'z', coord = ds.quan(0.,'code_length'))
# make a frb and save it to disk
center = (ds.quan(0.,'code_length'), ds.quan(0.,'code_length'))
xax,yax = ds.coordinates.x_axis[slc.axis], ds.coordinates.y_axis[slc.axis]
res = [ds.domain_dimensions[xax],ds.domain_dimensions[yax]]   # = [8,9]
width = ds.domain_right_edge[xax] - ds.domain_left_edge[xax]  # = 8 code_length
height = ds.domain_right_edge[yax] - ds.domain_left_edge[yax] # = 9 code_length
frb = slc.to_frb(width = width, height = height, resolution = res,
                 center= center)
fname = 'test_frb_roundtrip.h5'
frb.save_as_dataset(fname, fields = ['density'])

expected_vals = arr[:,:,5].T
print("\nConfirmation that initial frb results are expected:",
      (expected_vals == frb['density'].v).all(),'\n')

# yt-reload:
reloaded_ds = yt.load(fname)
print('\nShape of reloaded data:', reloaded_ds.data['density'].shape)
print('Reloaded data matches original after reshaping:',
      (expected_vals == 
       reloaded_ds.data['density'].reshape((9,8))).all())

# h5py-reload:
with h5py.File(fname,'r') as f:
    a = np.array(f['grid']['density'])
    print('\nShape of data stored on HDF5:', a.shape)
    print('Data on HDF5 match original values:',
          (expected_vals == a).all())
```

**Outcome**

```
yt : [INFO     ] 2020-08-13 15:40:20,073 Parameters: current_time              = 0.0
yt : [INFO     ] 2020-08-13 15:40:20,073 Parameters: domain_dimensions         = [ 8  9 10]
yt : [INFO     ] 2020-08-13 15:40:20,073 Parameters: domain_left_edge          = [-4.  -4.5 -5. ]
yt : [INFO     ] 2020-08-13 15:40:20,073 Parameters: domain_right_edge         = [4.  4.5 5. ]
yt : [INFO     ] 2020-08-13 15:40:20,074 Parameters: cosmological_simulation   = 0.0
yt : [INFO     ] 2020-08-13 15:40:20,196 Making a fixed resolution buffer of (('gas', 'density')) 8 by 9
yt : [INFO     ] 2020-08-13 15:40:20,198 Saving field data to yt dataset: test_frb_roundtrip.h5.
yt : [INFO     ] 2020-08-13 15:40:20,210 Making a fixed resolution buffer of (density) 8 by 9

Confirmation that initial frb results are expected: True 

yt : [INFO     ] 2020-08-13 15:40:20,302 Parameters: current_time              = 0.0 code_time
yt : [INFO     ] 2020-08-13 15:40:20,302 Parameters: domain_dimensions         = [8 9 1]
yt : [INFO     ] 2020-08-13 15:40:20,303 Parameters: domain_left_edge          = [-4.  -4.5  0. ] code_length
yt : [INFO     ] 2020-08-13 15:40:20,303 Parameters: domain_right_edge         = [4.  4.5 1. ] code_length
yt : [INFO     ] 2020-08-13 15:40:20,303 Parameters: cosmological_simulation   = 0.0

Shape of reloaded data: (8, 9)
Reloaded data matches original after reshaping: True

Shape of data stored on HDF5: (9, 8)
Data on HDF5 match original values: True
```

To be clear, the issue is that the reloaded data has the wrong shape. It should be `(9,8)`, not `(8,9)`.

**Version Information**
  * Operating System: Ubuntu 20.04
  * Python Version: 3.7.2
  * yt version: 3.6.0 (also tested with the development branch)
  * h5py version: 2.10.0
  * HDF5 library version: 1.10.4

For reference, I use `pyenv` to manage my python installation and used pip to install yt.

**Comment by neutrinoceros** [comment]
Hi @mabruzzo , thank you very much for this clear and detailed report !
This looks like a bug indeed, I think @brittonsmith should be able to help here.

**Comment by brittonsmith** [comment]
Hi, yes, the `ActiveDimensions` parameter needs to be set as `[NY, NX]` but was being set at `[NX, NY]`. Good catch! Issuing PR now.

**Comment by mabruzzo** [comment]
Great! Thanks for fixing it so quickly!

**Comment by brittonsmith** [comment]
Resolved in PR #2863.
